plaid/src/apis/github/repos.rs

```diff
@@ -104,4 +110,107 @@ impl Github {
+
+        match self.make_generic_get_request(address, module).await {
+            Ok((status, Ok(body))) => {
+                if status == 200 {
+                    Ok(body)
+                } else {
+                    Err(ApiError::GitHubError(GitHubError::UnexpectedStatusCode(
+                        status,
+                    )))
+                }
+            }
+            Ok((_, Err(e))) => Err(e),
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Get all collaborators on a repository.
+    /// See https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#list-repository-collaborators for more detail
+    pub async fn get_repository_collaborators(
+        &self,
+        params: &str,
+        module: &str,
+    ) -> Result<String, ApiError> {
+        let request: HashMap<&str, &str> =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+
+        let owner = self.validate_username(request.get("owner").ok_or(ApiError::BadRequest)?)?;
+        let repo =
+            self.validate_repository_name(request.get("repo").ok_or(ApiError::BadRequest)?)?;
+
+        info!("Fetching collaborators for [{repo}]");
+        let address = format!("/repos/{owner}/{repo}/collaborators");
+
+        match self.make_generic_get_request(address, module).await {
+            Ok((status, Ok(body))) => {
+                if status == 200 {
+                    Ok(body)
+                } else {
+                    Err(ApiError::GitHubError(GitHubError::UnexpectedStatusCode(
+                        status,
+                    )))
+                }
+            }
+            Ok((_, Err(e))) => Err(e),
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Update (or create) a branch protection rule.
+    /// See https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#update-branch-protection for more detail
+    pub async fn update_branch_protection_rule(
+        &self,
+        params: &str,
+        module: &str,
+    ) -> Result<u32, ApiError> {
+        let request: HashMap<&str, &str> =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+
+        let owner = self.validate_username(request.get("owner").ok_or(ApiError::BadRequest)?)?;
+        let repo =
+            self.validate_repository_name(request.get("repo").ok_or(ApiError::BadRequest)?)?;
+        let branch =
+            self.validate_branch_name(request.get("branch").ok_or(ApiError::BadRequest)?)?;
+        let body = request.get("body").ok_or(ApiError::BadRequest)?;
+
+        info!("Updating branch protection rules for branch [{branch}] in repo [{repo}]");
+        let address = format!("/repos/{owner}/{repo}/branches/{branch}/protection");
+
+        match self
+            .make_generic_put_request(address, Some(body), module)
+            .await
+        {
+            Ok((status, _)) => {
```

Comment:
The return code might be different when something is updated so this would fail in that circumstance. Double check that is handled correctly.

Reply:
Double checked - only returns 200 in success and then various 4XX on failure